spec/Appendix A -- Field Selection.md

```diff
@@ -0,0 +1,549 @@
+# Appendix A: Specification of FieldSelectionMap Scalar
+
+## Introduction
+
+This appendix focuses on the specification of the {FieldSelectionMap} scalar type. {FieldSelectionMap} is
+designed to express semantic equivalence between arguments of a field and fields within the result 
+type. Specifically, it allows defining complex relationships between input arguments and fields in
+the output object by encapsulating these relationships within a parsable string format. It is used
+in the `@is` and `@requires` directives.
+
+To illustrate, consider a simple example from a GraphQL schema:
+
+```graphql
+type Query {
+    userById(userId: ID! @is(field: "id")): User! @lookup
+}
+```
+
+In this schema, the `userById` query uses the `@is` directive with {FieldSelectionMap} to declare that
+the `userId` argument is semantically equivalent to the `User.id` field. 
+
+An example query might look like this:
+
+```graphql
+query {
+    userById(userId: "123") {
+        id
+    }
+}
+```
+
+Here, it is exptected that the `userId` "123" corresponds directly to `User.id`, resulting in the
+following response if correctly implemented:
+
+```json
+{
+    "data": {
+        "userById": {
+            "id": "123"
+        }
+    }
+}
+```
+
+The {FieldSelectionMap} scalar type is used to establish semantic equivalence between an argument and
+the fields within the associated return type. To accomplish this, the scalar must define the
+relationship between input fields or arguments and the fields in the resulting object. 
+Consequently, a {FieldSelectionMap} only makes sense in the context of a specific argument and its 
+return type.
+
+The {FieldSelectionMap} scalar is represented as a string that, when parsed, produces a {SelectedValue}.
+
+A {SelectedValue} must exactly match the shape of the argument value to be considered
+valid. For non-scalar arguments, you must specify each field of the input type in
+{SelectedObjectValue}.
+
+```graphql example
+extend type Query {
+    findUserByName(user: UserInput! @is(field: "{ firstName: firstName }")): User @lookup
+}
+```
+
+```graphql counter-example
+extend type Query {
+    findUserByName(user: UserInput! @is(field: "firstName")): User @lookup
```

Comment:
I agree this example is invalid, because the result is a scalar and not an object value. But I still think we should allow a shorthand when the field names match, so `{ firstName }` *would* be valid (similar to how JavaScript allows shorthand property definitions in object literals).